platform/darwin/src/MGLShapeSource.mm

```diff
@@ -56,6 +56,21 @@ - (instancetype)initWithIdentifier:(NSString *)identifier shape:(nullable MGLSha
     return self;
 }
 
+- (instancetype)initWithIdentifier:(NSString *)identifier features:(NS_ARRAY_OF(id <MGLFeature>) *)features options:(nullable NS_DICTIONARY_OF(MGLShapeSourceOption, id) *)options {
+    for (id <MGLFeature> feature in features) {
+        if (![feature conformsToProtocol:@protocol(MGLFeature)]) {
+            [NSException raise:NSInvalidArgumentException format:@"The object %@ included in the features argument does not conform to the MGLFeature protocol.", feature];
+        }
+    }
+    MGLShapeCollectionFeature *shapeCollectionFeature = [MGLShapeCollectionFeature shapeCollectionWithShapes:(NSArray *)features];
```

Comment:
You likely had to cast here because `+[MGLShapeCollectionFeature shapeCollectionWithShapes:]` accepts an array of `MGLShape <MGLFeature> *` instead of an array of `id <MGLFeature>`. Let’s make this method and `+[MGLShapeCollectionFeature shapeCollectionWithShapes:]` consistent with each other. (I have no strong preference one way or the other, but `id <MGLFeature>` might leave more options open for fixing #7454.)